server/bundles/io.cloudbeaver.service.ldap.auth/src/io/cloudbeaver/service/ldap/auth/LdapAuthProvider.java

```diff
@@ -185,15 +194,48 @@ protected String getAttributeValueSafe(@NotNull Attributes attributes, @NotNull
     }
 
     @NotNull
-    public Hashtable<String, String> creteAuthEnvironment(LdapSettings ldapSettings) {
+    public Hashtable<String, String> creteAuthEnvironment(LdapSettings ldapSettings) throws DBException {
         Hashtable<String, String> environment = new Hashtable<>();
         environment.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
 
         environment.put(Context.PROVIDER_URL, ldapSettings.getLdapProviderUrl());
         environment.put(Context.SECURITY_AUTHENTICATION, "simple");
+
+        try {
+            configureSsl(ldapSettings, environment);
+        } catch (Exception e) {
+            log.error("Can't establish ssl connection", e);
+            throw new DBException("Can't establish ssl connection", e);
+        }
+
         return environment;
     }
 
+    private void configureSsl(LdapSettings ldapSettings, Hashtable<String, String> environment) throws Exception {
```

Comment:
Hashtable??
LEt's use Map everywhere and convert it into hashtable for legacy API calls only